airflow/providers/google/cloud/triggers/kubernetes_engine.py

```diff
@@ -237,3 +243,67 @@ def _get_hook(self) -> GKEAsyncHook:
                 impersonation_chain=self.impersonation_chain,
             )
         return self._hook
+
+
+class GKEJobTrigger(BaseTrigger):
+    """GKEJobTrigger run on the trigger worker to check the state of Job."""
+
+    def __init__(
+        self,
+        cluster_url: str,
+        ssl_ca_cert: str,
+        job_name: str,
+        job_namespace: str,
+        gcp_conn_id: str = "google_cloud_default",
+        poll_interval: float = 2,
+        impersonation_chain: str | Sequence[str] | None = None,
+    ):
+        super().__init__()
+        self.cluster_url = cluster_url
+        self.ssl_ca_cert = ssl_ca_cert
+        self.job_name = job_name
+        self.job_namespace = job_namespace
+        self.gcp_conn_id = gcp_conn_id
+        self.poll_interval = poll_interval
+        self.impersonation_chain = impersonation_chain
+
+    def serialize(self) -> tuple[str, dict[str, Any]]:
+        """Serialize KubernetesCreateJobTrigger arguments and classpath."""
+        return (
+            "airflow.providers.google.cloud.triggers.kubernetes_engine.GKEJobTrigger",
+            {
+                "cluster_url": self.cluster_url,
+                "ssl_ca_cert": self.ssl_ca_cert,
+                "job_name": self.job_name,
+                "job_namespace": self.job_namespace,
+                "gcp_conn_id": self.gcp_conn_id,
+                "poll_interval": self.poll_interval,
+                "impersonation_chain": self.impersonation_chain,
+            },
+        )
+
+    async def run(self) -> AsyncIterator[TriggerEvent]:  # type: ignore[override]
+        """Get current job status and yield a TriggerEvent."""
+        job: V1Job = await self.hook.wait_until_job_complete(name=self.job_name, namespace=self.job_namespace)
+        job_dict = job.to_dict()
+        error_message = self.hook.is_job_failed(job=job)
+        yield TriggerEvent(
+            {
+                "name": job.metadata.name,
+                "namespace": job.metadata.namespace,
+                "status": "error" if error_message else "success",
+                "message": f"Job failed with error: {error_message}"
+                if error_message
+                else "Job completed successfully",
+                "job": job_dict,
+            }
+        )
```

Comment:
```suggestion
        status = "error" if error_message else "success"
        message = f"Job failed with error: {error_message}" if error_message else "Job completed successfully"
        yield TriggerEvent(
            {
                "name": job.metadata.name,
                "namespace": job.metadata.namespace,
                "status": status
                "message": message,
                "job": job_dict,
            }
        )
```

I would suggest we move the if-else block out of the dict for better readbility